markdown/App.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+    <div id="MarkdownApp">
+        <div v-if="isLoading">LOADING</div>
+        <MarkdownView v-if="isRendering" v-bind:markdown="markdown" v-on:edit="toggle" />
+        <MarkdownEdit v-if="isEditing" v-bind:markdown="markdown" v-on:save="toggle" />
+    </div>
+</template>
+
+<script lang="ts">
+  import { Component, Prop, Vue } from 'vue-property-decorator';
+  import MarkdownView from './components/MarkdownView.vue'
+  import { NamedNode } from 'rdflib'
+  import { loadMarkdown, saveMarkdown, STATE } from './markdown.service'
+  import MarkdownEdit from './components/MarkdownEdit.vue'
+
+  @Component({
+    components: {
+      MarkdownEdit,
+      MarkdownView
+    }
+  })
+  export default class App extends Vue {
+    state: STATE = STATE.LOADING
+    markdown: string = ''
+
+    @Prop(NamedNode) subject!: NamedNode
```

Comment:
If I don't use the exclamation mark I have to write `NamedNode | undefined` - this would be more correct if I couldn't guarantee subject being set before the created event (where I need it), but I can guarantee it so thought it was better to use the exclamation mark.

Reply:
Ah, I see - because you're declaring but not initialising the property, TS thinks it could be undefined. (And theoretically it can be, if you don't pass the prop in `index.ts`'s `render` function. But we do, and its type does not include `undefined` there; so yeah, it's justified.